### PR TITLE
Update openblas check for v1.7

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
@staticfloat @yingboma

### CODE REVIEW

File: src/linear_nonlinear.jl [hunk 1]
```diff
@@ -71,15 +71,22 @@ Base.resize!(p::LinSolveGPUFactorize,i) = p.A = nothing
...
   if !(reltol isa Real)
     error("Non real valued reltol is not supported by the linear iterative solvers. To customize tolerances for the linear iterative solvers, use the syntax like `KenCarp3(linsolve=LinSolveGMRES(abstol=1e-16,reltol=1e-16))`.")
   end
   return reltol
 end
 
+function isopenblas()
+    splitter = Base.Sys.iswindows() ? "\\" : "/"
+    str = split(BLAS.get_config().loaded_libs[1].libname,splitter)[end]
+    str == "libopenblas64_.dll" || str == "libopenblas_.dll"
```
Comment (YingboMa) [inline src/linear_nonlinear.jl:87]:
```suggestion
    occursin("openblas", str)
```

File: src/linear_nonlinear.jl [hunk 2]
```diff
@@ -71,15 +71,22 @@ Base.resize!(p::LinSolveGPUFactorize,i) = p.A = nothing
 mutable struct DefaultLinSolve
   A
   iterable
+  openblas
```
Comment (YingboMa) [inline src/linear_nonlinear.jl:74]:
```suggestion
  openblas::Bool
```

Comment (ChrisRackauckas) [reply src/linear_nonlinear.jl]: the initialization couldn't be nothing then. It should be Union{Nothing,Bool}

File: src/linear_nonlinear.jl [hunk 3]
```diff
@@ -71,15 +71,22 @@ Base.resize!(p::LinSolveGPUFactorize,i) = p.A = nothing
 mutable struct DefaultLinSolve
   A
   iterable
+  openblas::Bool
```
Comment (YingboMa) [inline src/linear_nonlinear.jl:74]:
```suggestion
  openblas::Union{Bool, Nothing}
```

File: src/linear_nonlinear.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/linear_nonlinear.jl]: wait until someone make a BLAS called "atleastitsbetterthanopenblas".

**Comment by YingboMa** [comment]
Remember to squash when merging it :p

**Comment by ChrisRackauckas** [comment]
This can't merge until v1.7 release, so we should wait a bit, but at least it's ready.

**Comment by YingboMa** [comment]
Or, we could just add a version check so this is backward compatible.

**Comment by ChrisRackauckas** [comment]
true, I'll do that.

File: src/linear_nonlinear.jl [hunk 4]
```diff
@@ -82,9 +82,14 @@ DefaultLinSolve() = DefaultLinSolve(nothing, nothing, nothing)
 end
 
 function isopenblas()
-    splitter = Base.Sys.iswindows() ? "\\" : "/"
-    str = split(BLAS.get_config().loaded_libs[1].libname,splitter)[end]
-    occursin("openblas", str)
+    if VERSION < v"1.7"
```
Comment (YingboMa) [inline src/linear_nonlinear.jl:85]:
```suggestion
    @static if VERSION < v"1.7"
```

File: src/linear_nonlinear.jl [hunk 5]
```diff
@@ -71,15 +71,27 @@ Base.resize!(p::LinSolveGPUFactorize,i) = p.A = nothing
...
   return reltol
 end
 
+function isopenblas()
+    if VERSION < v"1.7"
+      blas = BLAS.vendor()
+      blas == :openblas64 || blas == :openblas
+    else
+      splitter = Base.Sys.iswindows() ? "\\" : "/"
+      str = split(BLAS.get_config().loaded_libs[1].libname,splitter)[end]
```
Comment (YingboMa) [inline src/linear_nonlinear.jl:90]: With `occursin`, you won't need to use `split`

Comment (ChrisRackauckas) [reply src/linear_nonlinear.jl:90]: oops

**Review by YingboMa — APPROVED** [review]

File: Project.toml [hunk 6]
```diff
@@ -58,7 +58,7 @@ SciMLBase = "1.13"
 Setfield = "0.7"
 StaticArrays = "0.11, 0.12, 1.0"
 ZygoteRules = "0.2"
-julia = "1.6"
+julia = "1.7"
```
Comment (YingboMa) [inline Project.toml:61]: This change probably isn't needed anymore

Comment (ChrisRackauckas) [reply Project.toml:61]: true

**Comment by YingboMa** [comment]
Let's wait the release of 1.7 so that the CI can properly test it.

**Comment by ChrisRackauckas** [comment]
ehh I've been testing it locally in its various forms for days on a few different systems. It should at least be fine enough to release to users who are on the beta.

**Pull request merged**